E0365 needs to be updated to new format

From: src/test/compile-fail/E0365.rs

E0365 needs a span_label and a span_note changed to a note, updating it from:

```
error[E0365]: `foo` is private, and cannot be reexported
  --> src/test/compile-fail/E0365.rs:15:9
   |
15 | pub use foo as foo2; //~ ERROR E0365
   |         ^^^^^^^^^^^
   |
note: consider declaring type or module `foo` with `pub`
  --> src/test/compile-fail/E0365.rs:15:9
   |
15 | pub use foo as foo2; //~ ERROR E0365
   |         ^^^^^^^^^^^
```

To:

```
error[E0365]: `foo` is private, and cannot be reexported
  --> src/test/compile-fail/E0365.rs:15:9
   |
15 | pub use foo as foo2; //~ ERROR E0365
   |         ^^^^^^^^^^^ reexport of private `foo`
   |
   = note: consider declaring type or module `foo` with `pub`
```

--- a/src/librustc_resolve/resolve_imports.rs
+++ b/src/librustc_resolve/resolve_imports.rs
@@ -582,12 +582,12 @@ impl<'a, 'b:'a> ImportResolver<'a, 'b> {
                                        source);
                     self.session.add_lint(PRIVATE_IN_PUBLIC, directive.id, directive.span, msg);
                 } else {
-                    let msg = format!("`{}` is private, and cannot be reexported", source);
-                    let note_msg =
-                        format!("consider declaring type or module `{}` with `pub`", source);
-                    struct_span_err!(self.session, directive.span, E0365, "{}", &msg)
-                        .span_note(directive.span, &note_msg)
-                        .emit();
+                    let mut err = struct_span_err!(self.session, directive.span, E0365,
+                                                     "`{}` is private, and cannot be reexported",
+                                                     source);
+                    err.span_label(directive.span, &format!("reexport of private `{}`", source));
+                    err.note(&format!("consider declaring type or module `{}` with `pub`", source));
+                    err.emit();
                 }
             }
 

--- a/src/test/compile-fail/E0365.rs
+++ b/src/test/compile-fail/E0365.rs
@@ -12,6 +12,9 @@ mod foo {
     pub const X: u32 = 1;
 }
 
-pub use foo as foo2; //~ ERROR E0365
+pub use foo as foo2;
+//~^ ERROR `foo` is private, and cannot be reexported [E0365]
+//~| NOTE reexport of private `foo`
+//~| NOTE consider declaring type or module `foo` with `pub`
 
 fn main() {}

